java-cfenv-boot/src/main/java/io/pivotal/cfenv/spring/boot/GenAIModelInfo.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pivotal.cfenv.spring.boot;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents information about a GenAI model including its name and capabilities.
+ *
+ * @author Corby Page
+ */
+public class GenAIModelInfo {
+
+    public enum Capability {
```

Comment:
I personally wouldn't make this an enum.  We're expecting the capabilities returned by the ai-server to increase over time.  If this is an enum we need to ensure this is updated in lock-step with ai-server.

Reply:
OK, I have made Capability a non-enum class.